### ISSUE TITLE
Queue doesn't update/change if contacts removed from campaign

### ISSUE THREAD

**Issue by Lalilu** [mautic/mautic#6405]
**Please DO NOT report security vulnerabilities here. Send them to security@mautic.com instead.**

What type of report is this:

| Q  | A
| ---| ---
| Bug report? | X
| Feature request? | 
| Enhancement? | 

## Description:
If campaign is set up and mails are scheduled, removing contacts from the campaign (or setting up "Do Not Contact") doesn't have an effect on queue.

At least it is not visually represented. There should be lower numbers in all stages besides the lower one:
<img width="420" alt="grafik" src="https://user-images.githubusercontent.com/9094193/43474503-66f97a44-94f3-11e8-92cc-ad16006a2723.png">

## If a bug:

| Q   | A
| --- | ---
| Mautic version | v2.13.1
| PHP version | 5.6.32-nmm1

### Steps to reproduce:
1. Create a basho-like campaign with mails to be sent with a bit of delay between them 
2. Fill campaign with contacts
3. Publish it (shoot!)
4. Make a contact "Do Not Contact" or remove someone from the campaign
5. See that the next mails still have the same number of mails to be send (see screenshot above, 3XXX)

Any ideas?

**Comment by alanhartless** [comment]
Do not contact does not affect contacts because it is event specific and not campaign specific. Those counts are campaign aware meaning they do not have any relationship to what the action itself is supposed to do (email and dnc status). DNC will be evaluated when the event is executed for the contact.  If you want DNC to be accounted for in the campaign itself, you need to adjust your segment to exclude those so that they are removed from the campaign once they are marked as DNC. 

2.14.0 completely rewrites the backend of campaigns and campaign membership. Removed contacts still getting accounted for in the campaign progress bars will need to be confirmed as a bug in 2.14 before anything further is addressed.

**Comment by npracht** [comment]
Hi @Lalilu according to Alan's message, please upgrade and share if this should be addressed as a bug or not.
Thanks !

**Comment by Lalilu** [comment]
Hey guys!

Thanks for your explanations and encouragement to update! 😀 

I updated and now hope that everything works fine. Just saw the notifications about Do-not-contact Contacts which weren't successfully addressed. Which is good! :)

**Removed contacts still getting accounted for in the campaign progress bars** is, as it seems, the only existing bug.

**Comment by Lalilu** [comment]
@alanhartless I have a follow-up question on, a bit unrelated with the DNC handling:

If a campaign is deactivated, how do I make sure any mails (even previously scheduled mails) are not being sent? Contacts are not DNC, and will not be (as long as they don't bounce or unsubscribe).

---

Optional context: To investigate why a recent campaign didn't send mails (campaign event (to send a mail) was triggered, but "email sent" wasn't shown in contact's logs), I cloned campaigns and deactivated the old ones. But mails are still scheduled.

```
2nd mail (48 hours) event will be triggered at 180805 Traders List C new
August 9, 2018 9:47 pm CEST

2nd mail (48 hours) event will be triggered at 180805 Traders List C new2
August 9, 2018 11:06 pm CEST

2nd mail (48 hours) event will be triggered at 180805 Traders List C new3
August 10, 2018 11:58 am CEST
```

I now found out that it was simply to low server ressources, so I added `--batch-limits` to relevant cron jobs. But sadly, now I have e.g. 3 same mails scheduled, 2 of them are from old, now deactivated/unpublished campaigns.
